packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -379,6 +385,12 @@ export async function activate(context: vscode.ExtensionContext) {
 
 export function deactivate(): Promise<void> {
   console.log('SFDX CLI Extension Deactivated');
-  disposeTraceFlagExpiration();
+
+  decorators.disposeTraceFlagExpiration();
+
+  if (isDemoMode()) {
+    forceAuthLogoutAll();
```

Comment:
I think you might need to put this and restoreDebugLevels in a Promise.all and return that promise. The deactivate method will wait 4 seconds or until the returned promise is completed so forceAuthLogoutAll might not be allowed to complete if restoreDebugLevels finishes before.

https://github.com/Microsoft/vscode/issues/11895

Reply:
Thanks. I'm thinking that It's likely that I am going to remove this. The 4 seconds is too finicky. 

With demo mode the user is also expected to use a guest account on the OS, which will wipe out all data after that. That is a safer guarantee. This 4 second wait is not very reliable.